tests/scenario/test_grafana_cloud_integrator_lib.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+from charms.grafana_cloud_integrator.v0.cloud_config_requirer import (
+    CloudConfigAvailableEvent,
+    CloudConfigRevokedEvent,
+    GrafanaCloudConfigRequirer,
+)
+from ops import CharmBase, Framework
+from scenario import Context, Relation, State
+
+
+class MyCharm(CharmBase):
+    def __init__(self, framework: Framework):
+        super().__init__(framework)
+        self.cloud = GrafanaCloudConfigRequirer(self)
+
+
+@pytest.fixture()
+def context():
+    return Context(
+        charm_type=MyCharm,
+        meta={
+            "name": "my-charm",
+            "requires": {
+                "grafana-cloud-config": {"interface": "grafana_cloud_config", "limit": 1}
+            },
+        },
+    )
+
+
+@pytest.mark.parametrize(
+    "is_leader",
+    [
+        (True,),
+        (False,),
+    ],
+)
+def test_requirer_emits_cloud_config_available_event_on_relation_changed(is_leader, context):
+    # GIVEN a grafana-cloud-config relation
+    grafana_cloud_config_relation = Relation("grafana-cloud-config")
+    # AND GIVEN leadership/non-leadership
+    state = State(leader=is_leader, relations=[grafana_cloud_config_relation])
+
+    # WHEN the grafana-cloud-config relation changes
+    context.run(grafana_cloud_config_relation.changed_event, state)
+
+    # THEN the CloudConfigAvailableEvent event is emitted
+    any(event for event in context.emitted_events if isinstance(event, CloudConfigAvailableEvent))
```

Comment:
Assertion missing?

Reply:
Yeah that is a terrible omission.  Fixed